[docs][Ray Clusters] New Ray Clusters getting started page. (#27391)
ray-project/ray: doc/source/cluster/cluster_under_construction/getting-started.rst

----------------------

One of Ray's strengths is the ability to leverage multiple machines for
distributed execution. Ray is great for multiprocessing on a single machine.
However, the real power of Ray is the ability to seamlessly scale to a cluster

> of machines

Reviewer: ```suggestion
of machines.
```

A Ray cluster is a set of one or more nodes that are running Ray and share the same :ref:`head node<cluster-head-node-under-construction>`.
Ray clusters can either be a fixed-size number of nodes or :ref:`can autoscale<cluster-autoscaler-under-construction>` (i.e., automatically provision or deprovision the number of nodes in a cluster) according to the demand of the Ray workload.

How can I use Ray clusters?
~~~~~~~~~~~~~~~~~~~~~~~~~~~
Ray clusters are officially supported on the following technology stacks:

* The :ref:`Ray Cluster Launcher on AWS and GCP<ref-cluster-quick-start-vms-under-construction>`. Community-supported Azure and Aliyun integrations also exist.
* :ref:`KubeRay, the official way to run Ray on Kubernetes<kuberay-index>`.

Advanced users may want to :ref:`deploy Ray clusters on-premise<cluster-private-setup-under-construction>` or even onto infrastructure platforms not listed here by :ref:`providing a custom node provider<additional-cloud-providers-under-construction>`.

Where to go from here?
----------------------

.. panels::
    :container: text-center
    :column: col-lg-6 px-3 py-2
    :card:

>     **I want to learn key Ray concepts**

Reviewer: ```suggestion
    **I want to learn key Ray cluster concepts**
```

> distributed execution. Ray can, of course, be run on a single machine (and is
> done so often), but the real power is using Ray on a cluster of machines.

Reviewer: ```suggestion
distributed execution. Ray can be run on a single machine and is
done so often, but the real power of Ray is the ability to seamlessly scale to a cluster of machines.
```

Reviewer 2: This is still ever-so-slightly ungrammatical.
How about:
`Ray is great for multiprocessing on a single machine. However, the real power of Ray is the ability to seamlessly scale to a cluster of machines.`

Reviewer: good catch... didn't read that paren phrase well

A Ray cluster is a set of one or more nodes that are running Ray and share the same :ref:`head node<cluster-head-node-under-construction>`.

> Ray clusters can be a fixed-size number of nodes, or :ref:`autoscale<cluster-autoscaler-under-construction>` the cluster size up and down according to the demand of the Ray workload.

Reviewer: ```suggestion
Ray clusters can either be a fixed-size number of nodes or :ref:`can autoscale<cluster-autoscaler-under-construction>` (i.e., automatically provision or deprovision the number of nodes in a cluster) according to the demand of the Ray workload.
```

distributed execution. Ray can, of course, be run on a single machine (and is
done so often), but the real power is using Ray on a cluster of machines.

> A Ray Cluster can be created from virtual machines using the :ref:`Ray Cluster Launcher<ref-cluster-quick-start-vms-under-construction>`, which is officially supported on AWS and GCP and community supported on Azure and Aliyun, or created from machines managed by Kubernetes via :ref:`KubeRay<kuberay-index>`, which is the officially supported way to deploy Ray on Kubernetes.

Reviewer 2: This sentence is long. It should be broken up.

Ray can also be :ref:`deployed on-premise<cluster-private-setup-under-construction>` or even onto infrastructure not listed here by :ref:`providing a custom node provider<additional-cloud-providers-under-construction>`.

> Your cluster can have a fixed size, or :ref:`leverage autoscaling<cluster-autoscaler-under-construction>` to automatically request additional or release unused nodes depending on the demands of your application.

Reviewer 2: Nit/suggestion: `to automatically request additional resources or release unused nodes, according to the demands of your application.`

>     :column: col-lg-6 px-2 py-2

Reviewer: can we do px-3 here?

    :card:

    **I want to run Ray on a cloud provider** 
    ^^^
    This guide helps you take a sample application designed to run on a laptop and
    scale it up in the cloud. Access to an AWS or GCP account is required.

    +++
    .. link-button:: ref-cluster-quick-start-vms-under-construction
        :type: ref
        :text: Getting Started with Ray Clusters on VMs
        :classes: btn-outline-info btn-block
    ---

    **I want to run Ray on Kubernetes**
    ^^^
    This guide helps you deploy a Ray application to a Kubernetes cluster.
    You can run the tutorial on a remote Kubernetes cluster or on your laptop via KinD.

    +++
    .. link-button:: kuberay-quickstart
        :type: ref
        :text: Getting Started with Ray on Kubernetes
        :classes: btn-outline-info btn-block
    ---

>     **I want to learn key Ray concepts**

Reviewer: should this be first on the list?

    ^^^
    Understand the key concepts behind Ray Clusters. Learn about the main

>     concepts and the different ways to interact with a cluster.

Reviewer 2: This description could be made less redundant -- for example, the word "concepts" should appear just once.

Reviewer 2: (Covered by Richard's suggestion below.)

>     Understand the key concepts behind Ray Clusters. Learn about the main
>     concepts and the different ways to interact with a cluster.

Reviewer: ```suggestion
    Understand the key concepts and main ways of interacting with a Ray cluster.
```

    :column: col-lg-6 px-2 py-2
    :card:

>     **Getting Started with the Ray Clusters on Virtual Machines** 

Reviewer 2: there's a spare `the`

    ^^^

    In this quick start tutorial you will take a sample application designed to
    run on a laptop and scale it up in the cloud, using a cloud provider such as
    AWS or GCP.

    +++
    .. link-button:: ref-cluster-quick-start-vms-under-construction
        :type: ref
        :text: Getting Started with Ray Clusters on VMs
        :classes: btn-outline-info btn-block
    ---

    **Getting Started with Ray on Kubernetes**
    ^^^

    In this quick start tutorial you will take a sample application designed to
    run on a laptop and scale it up in the cloud, using a cloud provider such as

>     AWS or GCP.

Reviewer 2: Suggestion based on the content of that quickstart:
```
You will learn how to deploy a Ray application to a Kubernetes cluster.
You can run the tutorial on a remote Kubernetes cluster or on your laptop via KinD.
```
